kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManager.java

```diff
@@ -101,12 +102,16 @@ private static ExecutorService queryThreadPool() {
     // CompletableFuture.allOf and converting the CompletableFuture<Void> to
     // CompletableFuture<List<SearchResult>>
     CompletableFuture<List<SearchResult<T>>> searchResults = CompletableFutures.allAsList(queries);
-
-    // Increment the node count right at the end so that we increment it only once
-    //noinspection unchecked
-    return ((SearchResultAggregator<T>) new SearchResultAggregatorImpl<>(query))
-        .aggregate(searchResults)
-        .thenApply(this::incrementNodeCount);
+    try {
+      //noinspection unchecked
+      SearchResult<T> aggregatedResults =
+          ((SearchResultAggregator<T>) new SearchResultAggregatorImpl<>(query))
+              .aggregate(searchResults.get(30, TimeUnit.SECONDS));
+      return incrementNodeCount(aggregatedResults);
+    } catch (Exception e) {
+      LOG.error("Error searching across chunks ", e);
+      throw new IOException(e);
```

Comment:
Do we need to throw a checked exception here?

Reply:
If we throw Runtime exception how would you suggest catching it in `KaldbQueryServiceBase` ?